lib/repl.js

```diff
@@ -585,19 +595,20 @@ REPLServer.prototype.createContext = function() {
     context = global;
   } else {
     context = vm.createContext();
+    context.global = context;
+    context.global.global = context;
     const _console = new Console(this.outputStream);
-    Object.getOwnPropertyNames(global).forEach((name) => {
-      if (name === 'console') {
-        Object.defineProperty(context, name, {
-          configurable: true,
-          enumerable: true,
-          value: _console
-        });
-      }
-      else {
-        Object.defineProperty(context, name,
-          Object.getOwnPropertyDescriptor(global, name));
-      }
+    Object.defineProperty(context, 'console', {
+      configurable: true,
+      enumerable: true,
+      get: () => _console
+    });
+    Object.getOwnPropertyNames(global).filter((name) => {
+      if (name === 'console' || name === 'global') return false;
+      return GLOBAL_OBJECT_PROPERTY_MAP[name] === undefined;
+    }).forEach((name) => {
+      Object.defineProperty(context, name,
```

Comment:
@lance `instanceof` test for this filtered list may fail!. e.g `Int16Array`. Can you confirm that? Adding those missing properties to `GLOBAL_OBJECT_PROPERTIES` will solve this issue.